graylog2-web-interface/src/logic/datetimes/DateTime.ts

```diff
@@ -46,6 +46,10 @@ class DateTime {
     return new DateTime(moment.utc(dateTime));
   }
 
+  static fromDateTimeAndTZ(dateTime, timezone) {
+    return new DateTime(moment(dateTime).tz(timezone));
```

Comment:
This way it is actually parsing `dateTime` in the current timezone, instead of UTC. If e.g. the timezone is UTC, it will subtract the difference from the local timezone to UTC twice.

The effect of this is that there is a bug in the keyword time range tab:

  - Login as user with a UTC timezone configured (e.g. `admin`)
  - Open keyword time range tab
  - Enter e.g. `last week`
  - From/To dates will be shifted by 2 hours

A possible fix would look like this:
```suggestion
    return new DateTime(moment.utc(dateTime).tz(timezone));
```

Reply:
thanks for spotting this! I double checked.